great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -325,6 +325,29 @@ def _check_connection_string(cls, values: dict) -> dict:
             "Must provide either a connection string or a combination of account, user, and password."
         )
 
+    @pydantic.validator("assets", pre=True)
+    def _asset_forward_compatibility(cls, assets: list[dict]) -> list[dict]:
+        """
+        This validator is here to maintain compatibility with future versions of GX.
+        """
+        modified_assets: list[str] = []
+        try:
+            # if the incoming asset has a database_name, item we need to remove it.
+            # future versions of GX will support a `database_name`.
+            if assets:
+                for asset in assets:
+                    database = asset.pop("database", None)
+                    database_name = asset.pop("database_name", None)
+                    if asset_name := asset.get("name") and (database or database_name):
+                        modified_assets.append(asset_name)
+            if modified_assets:
+                LOGGER.info(
+                    f"Assets modified for forward compatibility: {', '.join(modified_assets)}"
+                )
```

Comment:
I debated whether this should be an actual `warning` prompting them to consider upgrading.